yaksh/views.py

```diff
@@ -612,37 +611,71 @@ def check(request, q_id, attempt_num=None, questionpaper_id=None):
             if current_question.type == 'code' or \
             current_question.type == 'upload' else None
         result = paper.validate_answer(
-            user_answer, current_question, json_data
+            user_answer, current_question, json_data, uid
         )
-        if result.get('success'):
-            new_answer.marks = (current_question.points * result['weight'] /
-                current_question.get_maximum_test_case_weight()) \
-                if current_question.partial_grading and \
-                current_question.type == 'code' or current_question.type == 'upload' \
-                else current_question.points
-            new_answer.correct = result.get('success')
-            error_message = None
-            new_answer.error = json.dumps(result.get('error'))
-            next_question = paper.add_completed_question(current_question.id)
+        if current_question.type in ['code', 'upload']:
+            if paper.time_left() <= 0:
+                url = 'http://localhost:%s' % SERVER_POOL_PORT
+                result = get_result_from_code_server(url, uid, block=True)
+                result = json.loads(result.get('result'))
```

Comment:
Avoid using the same variable name again and again. variable in line 620 can be <code>json_result</code> or something similar.

Reply:
Resolved